modules/ipc/include/hephaestus/ipc/zenoh/action_server.h

```diff
@@ -0,0 +1,250 @@
+    } catch (const std::exception& ex) {
+      LOG(ERROR) << fmt::format("ActionServer (topic: {}): execute callback failed with exception: {}.",
+                                topic_config_.name, ex.what());
+      return ReplyT{};
+    }
+  }();
+  const auto response_topic = internal::getResponseServiceTopic(topic_config_);
+  // TODO: if stop_requested == true, should we communicate somehow that we terminate cause stop was
+  // requested?
+  const auto client_response =
+      callService<ReplyT, ActionServerRequestResponse>(*session_, response_topic, reply);
+  if (client_response.empty() ||
+      client_response.front().value.status != ActionServerRequestStatus::ACCEPTED) {
+    LOG(ERROR) << fmt::format("ActionServer (topic: {}): failed to send final response to client.",
+                              topic_config_.name);
+    // TODO: should we do something else?
+  }
+
+  is_running_ = false;
+}
+
+// ----------------------------------------------------------------------------------------------------------
+// --------- ActionServerClient Implementation --------------------------------------------------------------
+// ----------------------------------------------------------------------------------------------------------
+
+template <typename ReplyT>
+[[nodiscard]] auto
+handleFailure(const std::string& topic_name, const std::string& error_message,
+              ActionServerRequestStatus error) -> std::future<ActionServerResponse<ReplyT>> {
+  LOG(ERROR) << fmt::format("Failed to call action server (topic: {}): {}.", topic_name, error_message);
+  std::promise<ActionServerResponse<ReplyT>> promise;
+  promise.set_value({ ReplyT{}, error });
+  return promise.get_future();
+}
+
+template <typename RequestT, typename StatusT, typename ReplyT>
+auto callActionServer(SessionPtr session, const TopicConfig& topic_config, const RequestT& request,
+                      StatusUpdateCallback<StatusT>&& status_update_cb)
+    -> std::future<ActionServerResponse<ReplyT>> {
+  const auto server_responses =
+      callService<RequestT, ActionServerRequestResponse>(*session, topic_config, request);
+  if (server_responses.empty()) {
+    return handleFailure<ReplyT>(topic_config.name, "no response", ActionServerRequestStatus::INVALID);
+  }
+
+  if (server_responses.size() > 1) {
+    return handleFailure<ReplyT>(
+        topic_config.name,
+        fmt::format(
+            "received more than one response ({}), make sure the topic matches a single action server",
+            server_responses.size()),
+        ActionServerRequestStatus::INVALID);
+  }
+
+  const auto& server_response_status = server_responses.front().value.status;
+  if (server_response_status != ActionServerRequestStatus::ACCEPTED) {
+    return handleFailure<ReplyT>(
+        topic_config.name, fmt::format("request rejected {}", magic_enum::enum_name(server_response_status)),
+        server_response_status);
+  }
+
+  // TODO: figure out why we need this
```

Comment:
what would be the alternative?

Reply:
the comment got misplaced by the formatter, it was related to the nolint